src/enhance-props.ts

```diff
@@ -28,13 +28,8 @@ export default function enhanceProps(
   let className: string = props.className || ''
 
   for (const [property, value] of propsMap) {
-    if (value && typeof value === 'object') {
-      // Do not attempt to serialize style prop into a classname - preserve it and move on
-      if (property === 'style') {
-        preservedProps[property] = value
-        continue
-      }
-
+    // Do not attempt to serialize style prop into a classname - it will be preserved as a native prop below
+    if (value && typeof value === 'object' && property !== 'style') {
```

Comment:
i think im missing what this does

Reply:
Ah! So this block is what recursively expands/enhances props for objects, really any object under `selectors`:

```tsx
const FancyBox = ({ children, ...rest }) => (
  <Box
    style={{ objectFit: "cover" }}
    selectors={{
      "&:hover": {
        background: "red",
      },
      "&:focus": { boxShadow: "rgba(67, 90, 111, 0.3)" },
    }}
    {...rest}
  >
    {children}
  </Box>
);
```

The keys that are under `selectors` are objects that get serialized with their selectors appended and injected into the stylesheet. However, `style` is an object that we shouldn't be trying to process in `enhanceProps` - it's a native prop that we should just pass through for consumers to use if needed (i.e., for unsupported props like `objectFit` https://github.com/segmentio/evergreen/blob/master/src/avatar/src/Avatar.js#L101)

Previously, we were more or less ignoring any props that didn't have a `propEnhancer` and forwarding it through - including those that might have object values. After writing this out, there might be other native HTML attributes that accept objects that might be clobbered by this. I'm gonna do some TypeScript-fiddling to see if I can find a list of other props that accept an object.